catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/DynamicSchemaResolver.java

```diff
@@ -719,37 +768,45 @@ private boolean isPhoneticsEnabled(Map<String, Serializable> enabledFeatures) {
 
   private void addToFieldsCache(Set<AttributeDescriptor> descriptors) {
     for (AttributeDescriptor ad : descriptors) {
+      addToFieldsCache(ad);
+    }
+  }
 
-      AttributeFormat format = ad.getType().getAttributeFormat();
+  private void addToFieldsCache(AttributeDescriptor descriptor) {
 
-      fieldsCache.add(ad.getName() + schemaFields.getFieldSuffix(format));
+    AttributeFormat format = descriptor.getType().getAttributeFormat();
 
-      if (!getSpecialIndexSuffix(format).equals("")) {
-        fieldsCache.add(
-            ad.getName() + schemaFields.getFieldSuffix(format) + getSpecialIndexSuffix(format));
-      }
+    fieldsCache.add(descriptor.getName() + schemaFields.getFieldSuffix(format));
```

Comment:
✏️ Since all these concatenations are done in a loop, might want to consider StringBuilder

Reply:
I think I'm going to leave it as it is for now. I think using an SB would make the code a lot more difficult to read. But if other team members feel strongly about it, I will reconsider.